File: Java/FileSharing/app/src/main/java/com/ale/filesharingdemo/activities/StartupActivity.java
package com.ale.filesharingdemo.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;

import com.ale.filesharingdemo.R;
import com.ale.filesharingdemo.fragments.ConversationsFragment;
import com.ale.filesharingdemo.fragments.FilesFragment;
import com.ale.filesharingdemo.fragments.LoginFragment;

public class StartupActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_startup);

        setSupportActionBar(findViewById(R.id.app_toolbar));

        openLoginFragment();
    }

    public void openLoginFragment() {
        openFragment(new LoginFragment(), false);
    }

    public void openFilesFragment() {
        openFragment(new FilesFragment(), true);
    }

    public void openConversationsFragment() {
        openFragment(new ConversationsFragment(), true);
    }

    private void openFragment(Fragment fragment, boolean addToBackStack) {
        if (addToBackStack) {
            getSupportFragmentManager().beginTransaction().setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN).replace(R.id.fragment_container, fragment).addToBackStack(null).commit();
        } else {
            getSupportFragmentManager().beginTransaction().setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN).replace(R.id.fragment_container, fragment).commit();

        }
    }
}

File: Kotlin/Skeleton/app/src/main/java/com/ale/skeleton/activities/StartupActivity.kt
package com.ale.conversations.activities

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.ale.conversations.R
import com.ale.conversations.extensions.transaction
import com.ale.conversations.fragments.*
import com.ale.infra.manager.room.Room
import com.ale.infra.proxy.conversation.IRainbowConversation
import kotlinx.android.synthetic.main.activity_main.*

class StartupActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        openLoginFragment()
        setSupportActionBar(app_toolbar)
    }

    fun openLoginFragment() = supportFragmentManager.transaction {
        replace(R.id.fragment_container, LoginFragment())
    }
}
